.github/workflows/pipeline.yml

```diff
@@ -14,6 +14,7 @@ jobs:
   build_and_test:
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
```

Comment:
Any reason why we wouldn't want `fail-fast: true` here?

Reply:
Personal preference but so the pipeline does not stop as soon as it fails on one of the OS. Can help to catch problems on different OS without having to re-run the CI.